### PR TITLE
[cloud_firestore] Move cloud_firestore to its own subdir.

### PR DESCRIPTION
**Pull request by ditman**
## Description

Move `cloud_firestore` to `cloud_firestore/cloud_firestore`.

(This is required for federated implementations.)

Update docs to point to the new location, and bump version number.

## Related Issues

* https://github.com/flutter/flutter/issues/45293

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by ditman** [comment]
```
  warning - The include file ../../analysis_options.yaml in /tmp/cirrus-ci-build/packages/cloud_firestore/cloud_firestore/analysis_options.yaml cannot be found at analysis_options.yaml:1:11 - (include_file_not_found)
```

Oopsie!

File: packages/cloud_firestore/cloud_firestore/pubspec.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: cloud_firestore
 description: Flutter plugin for Cloud Firestore, a cloud-hosted, noSQL database with
   live synchronization and offline support on Android and iOS.
 author: Flutter Team <flutter-dev@googlegroups.com>
-homepage: https://github.com/FirebaseExtended/flutterfire/tree/master/packages/cloud_firestore
-version: 0.12.10+3
+homepage: https://github.com/FirebaseExtended/flutterfire/tree/master/packages/cloud_firestore/cloud_firestore
+version: 0.12.11
```
Comment (hterkelsen) [inline packages/cloud_firestore/cloud_firestore/pubspec.yaml]: nit: let's make this `0.12.10+4`

Comment (ditman) [reply packages/cloud_firestore/cloud_firestore/pubspec.yaml]: OK! Updating

Comment (ditman) [reply packages/cloud_firestore/cloud_firestore/pubspec.yaml]: -ed

**Pull request merged**